homeassistant/components/fireservicerota/sensor.py

```diff
@@ -0,0 +1,154 @@
+"""Sensor platform for FireServiceRota integration."""
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import ATTRIBUTION, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up FireServiceRota sensor based on a config entry."""
+    data = hass.data[DOMAIN][entry.entry_id]
+    entry_id = entry.entry_id
+    unique_id = entry.unique_id
+
+    async_add_entities(
+        [IncidentsSensor(data, entry_id, unique_id, "Incidents", "mdi:fire-truck")],
+        True,
+    )
+
+
+class IncidentsSensor(RestoreEntity):
+    """Representation of FireServiceRota incidents sensor."""
+
+    def __init__(self, data, entry_id, unique_id, name, icon):
+        """Initialize."""
+        self._data = data
+        self._entry_id = entry_id
+        self._unique_id = unique_id
+        self._name = name
+        self._icon = icon
+
+        self._enabled_default = True
+        self._available = True
+        self._device_class = None
+
+        self._state = None
+        self._state_attributes = {}
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def icon(self) -> str:
+        """Return the icon to use in the frontend."""
+        return self._icon
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this sensor."""
+        return f"{self._unique_id}_{self._name}"
```

Comment:
Can the name change?

Reply:
@Kane610 If a user renames an entity because he has more than one account (unlikely) where is the new name stored?

Reply:
No once the sensor is created it stays the same. in this case it's something like: "my@email-address.nl_Incidents" any suggestions for a change? I have done all others, and the code gets smaller and smaller...